Bottom: 131551fb9b47bfc7b909d72692fcc8dfb91ea942
Top:    35d5c157f5c994e1483f67deea52fb711f3c974a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-13 01:22:41 +0100

Fix some broken assumptions regarding 32-bit integer coordinates in polygon1.c


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 572a138..4598f3c 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2511,8 +2511,8 @@ poly_IniContour (PLINE * c)
   /* bzero (c, sizeof(PLINE)); */
   c->head.next = c->head.prev = &c->head;
 #warning BROKEN ASSUMPTION of 32 BIT
-  c->xmin = c->ymin = 0x7fffffff;
-  c->xmax = c->ymax = 0x80000000;
+  c->xmin = c->ymin = COORD_MAX;
+  c->xmax = c->ymax = -COORD_MAX - 1;
   c->is_round = FALSE;
   c->cx = 0;
   c->cy = 0;
@@ -2904,8 +2904,7 @@ poly_InsideContour (PLINE * c, Vector p)
   info.f = 0;
   info.p[0] = ray.X1 = p[0];
   info.p[1] = ray.Y1 = p[1];
-#warning BROKEN ASSUMPTION of 32 BIT
-  ray.X2 = 0x7fffffff;
+  ray.X2 = COORD_MAX;
   ray.Y2 = p[1] + 1;
   if (setjmp (info.env) == 0)
     r_search (c->tree, &ray, NULL, crossing, &info);
